src/sonic-build-hooks/scripts/buildinfo_base.sh

```diff
@@ -152,57 +152,130 @@ download_packages()
 {
     local parameters=("$@")
     declare -A filenames
+    declare -A SRC_FILENAMES
+    local url=
+    local real_version=
+    local SRC_FILENAME=
+    local DST_FILENAME=
+
     for (( i=0; i<${#parameters[@]}; i++ ))
     do
         local para=${parameters[$i]}
         local nexti=$((i+1))
-        if [[ "$para" == *://* ]]; then
+        if [[ "$para" == -o || "$para" == -O ]]; then
+            DST_FILENAME=${parameters[$nexti]}
+        elif [[ "$para" == *://* ]]; then
             local url=$para
             local real_version=
 
             # Skip to use the proxy, if the url has already used the proxy server
-            if [[ $url == ${URL_PREFIX}* ]]; then
+            if [[ ! -z "${URL_PREFIX}" && $url == ${URL_PREFIX}* ]]; then
                 continue
             fi
+            local result=0
+            WEB_CACHE_PATH=${PKG_CACHE_PATH}/web
+            mkdir -p ${WEB_CACHE_PATH}
```

Comment:
The line 176 and 177 to make cache path can be moved to before the line 187, when the folder needed, right?